doc/developer-guides/coding_guidelines.rst

```diff
@@ -3110,3 +3100,176 @@ Compliant example::
        // This is a comment
 
 
+Naming Convention
+*****************
+
+
+NC-01: Object-like MACRO shall be named with full upper case
+============================================================
+
+Compliant example::
+
+    #define MAX_CONFIG_NAME_SIZE        32U
+
+.. rst-class:: non-compliant-code
+
+   Non-compliant example::
+
+       #define max_config_name_size        32U
+
+
+NC-02: Mixed-use of lower case and upper case in function-like MACRO shall not be allowed
+=========================================================================================
+
+Function-like MACRO shall be named with either full lower case or full upper
+case. Mixed-use of lower case and upper case shall not be allowed.
+
+Compliant example::
+
+    #define max(x, y)        ((x) < (y)) ? (y) : (x)
+
+.. rst-class:: non-compliant-code
+
+   Non-compliant example::
+
+       #define Max(x, y)        ((x) < (y)) ? (y) : (x)
+
+
+NC-03: Data structures exposed to external components shall be named with prefix 'acrn'
+=======================================================================================
+
+The data structure types include struct, union, and enum.
+This rule applies to the data structure with all the following properties:
+
+a) The data structure is used by multiple modules;
+b) The corresponding resource is exposed to external components, such as SOS or
+   UOS;
+c) The name meaning is simplistic or common, such as vcpu or vm.
+
+Compliant example::
+
+    struct acrn_vcpu {
+            ...
+    };
+
+.. rst-class:: non-compliant-code
+
+   Non-compliant example::
+
+       struct vcpu {
+               ...
+       };
+
+
+NC-04: Data structures only used by hypervisor shall be named with prefix 'hv'
+==============================================================================
+
+The data structure types include struct, union, and enum.
+This rule applies to the data structure with all the following properties:
+
+a) The data structure is used by multiple modules;
+b) The corresponding resource is only used by hypervisor;
+c) The name meaning is simplistic or common, such as timer.
+
+Compliant example::
+
+    struct hv_timer {
+            ...
+    };
+
+.. rst-class:: non-compliant-code
+
+   Non-compliant example::
+
+       struct timer {
+               ...
+       };
+
+
+NC-05: Data structures only used by one module shall be named with the module name as prefix
```

Comment:
Is there a precedence hierarchy for these naming conventions?  What if I have a data structure used in only one module in the hypervisor?  Which rule prevails: using ``hv_`` prefix or the module name prefix?

Reply:
In rule NC-03, it mentions that `The data structure is used by multiple modules;`. 
So, in this case, we should use the module name prefix.